File: C++/Task12/n15854/n15854.cpp
// https://inf-ege.sdamgia.ru/problem?id=15854

#include <iostream>

using namespace std;

int main()
{
	string s = string(1, '1') + string(80, '0');
	while (s.find("10") != string::npos || s.find("1") != string::npos)
	{
		size_t pos = s.find("10");
		if (pos != string::npos)
		{
			s.replace(pos, 2, "001");
		}
		else
		{
			pos = s.find("1");
			if (pos != string::npos)
			{
				s.replace(pos, 1, "000");
			}
		}
	}
	std::cout << count(s.begin(), s.end(), '0') << std::endl;
	return 0;
}

File: Python/Task12/n15854.py
# https://inf-ege.sdamgia.ru/problem?id=15854

s = '1' + '0'*80
while '10' in s or '1' in s:
	if '10' in s:
		s = s.replace('10', '001', 1)
	else:
		if '1' in s:
			s = s.replace('1', '000', 1)
print(s.count('0'))